engine: Fix TestGetOrderbookAmountByNominal panic

Thanks to @cranktakular for reporting this:

```
--- FAIL: TestGetOrderbookAmountByNominal (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x141c5ce33]

goroutine 3039 [running]:
testing.tRunner.func1.2({0x142198ca0, 0x141f5f490})
        C:/Program Files/Go/src/testing/testing.go:1872 +0x3fc
testing.tRunner.func1()
        C:/Program Files/Go/src/testing/testing.go:1875 +0x683
panic({0x142198ca0?, 0x141f5f490?})
        C:/Program Files/Go/src/runtime/panic.go:783 +0x132
github.com/thrasher-corp/gocryptotrader/engine.TestGetOrderbookAmountByNominal(0xc0005cfdc0)
        C:/Users/emp/Documents/dev/gocryptotrader/engine/rpcserver_test.go:3189 +0xdd3
testing.tRunner(0xc0005cfdc0, 0x142512840)
        C:/Program Files/Go/src/testing/testing.go:1934 +0x1c4
created by testing.(*T).Run in goroutine 1
        C:/Program Files/Go/src/testing/testing.go:1997 +0x9dd
FAIL    github.com/thrasher-corp/gocryptotrader/engine  74.512s
FAIL
```

--- a/engine/rpcserver_test.go
+++ b/engine/rpcserver_test.go
@@ -61,7 +61,14 @@ const (
 	fakeExchangeName      = "fake"
 )
 
-var errExpectedTestError = errors.New("expected test error")
+var (
+	errExpectedTestError = errors.New("expected test error")
+	testExchangeCounter  common.Counter
+)
+
+func newUniqueFakeExchangeName() string {
+	return fmt.Sprintf("%s-%d", fakeExchangeName, testExchangeCounter.IncrementAndGet())
+}
 
 // fExchange is a fake exchange with function overrides
 // we're not testing an actual exchange's implemented functions
@@ -3141,7 +3148,7 @@ func TestGetOrderbookAmountByNominal(t *testing.T) {
 
 	exch.SetDefaults()
 	b := exch.GetBase()
-	uniqueFakeExchangeName := fmt.Sprintf("%s-%d", fakeExchangeName, time.Now().UnixNano())
+	uniqueFakeExchangeName := newUniqueFakeExchangeName()
 	b.Name = uniqueFakeExchangeName
 	b.Enabled = true
 
@@ -3226,7 +3233,7 @@ func TestGetOrderbookAmountByImpact(t *testing.T) {
 
 	exch.SetDefaults()
 	b := exch.GetBase()
-	uniqueFakeExchangeName := fmt.Sprintf("%s-%d", fakeExchangeName, time.Now().UnixNano())
+	uniqueFakeExchangeName := newUniqueFakeExchangeName()
 	b.Name = uniqueFakeExchangeName
 	b.Enabled = true
 

